api/include/opentelemetry/trace/span_id.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace opentelemetry {
+namespace trace {
+
+class SpanId {
+ public:
+  // The size in bytes of the SpanId.
+  static constexpr size_t kSize = 8;
+
+  // An invalid SpanId (all zeros).
+  SpanId() : rep_{0} {}
+
+  // Creates a SpanId by copying the first kSize bytes from the buffer.
+  explicit SpanId(const uint8_t* buf);
+
+  // Returns a 16-char hex string of the SpanId value.
+  std::string ToHex() const;
+
+ private:
+  uint8_t rep_[kSize];
```

Comment:
Any reason we're using an array of 8-bit numbers to represent this type instead of a uint64_t?

A char array like this will have a byte-level alignment (e.g. alignof(SpanId) == 1)) and will be less performant for doing basic operations like equality check, assignment, etc.

Reply:
I wanted to avoid implying that this is an integer, and thereby avoid endianness issues. SpanId is an opaque 8-byte binary blob.

Reply:
This has been discussed in the protocol repository, fwiw: https://github.com/open-telemetry/opentelemetry-proto/pull/52

(It looks like we'll end up on using a byte array in the protocol, which doesn't preclude using 64-bit integers here.)

Reply:
Can we defer this as an implementation detail, or do we treat alignment as part of the ABI, and therefore have to set it in stone?